### PR TITLE
Add more Planned Enhancements

### PR DESCRIPTION
**Pull request by yong-zhuo**
* Add planned enhancements related to improving `help` command and `NRIC` validation based on #374 and #397 
* Also fix formatting issues with proposed implementation of 'Undo/Redo' command

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1149,4 +1151,12 @@ Team size: 5
   Currently, all fields are mandatory when adding a patient. However, there could be cases where the patient does not have an email address or phone number.
   It would be ideal to make certain fields optional as it allows for more flexibility when adding patient details. However, the NRIC field should remain mandatory since it is the unique identifier for each patient which ensures that duplicate patients cannot be added.
 
+8\. **Include Command Overview in Help Pop-Up**
+
+  Currently, the pop-up that appears after using the `help` command only provides a link to the detailed user guide. Users are also required to copy the URL given, open a new browser and navigate away from the application to view the commands available.
+  We plan to include a command overview in the pop-up that will display a list of all available commands and their respective constraints. This will allow users to quickly view the commands available without having to navigate away from the application.
```
Comment (PresleyChew) [inline docs/DeveloperGuide.md:1157]: Caps "command overview" since you caps in header

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1149,4 +1151,12 @@ Team size: 5
...
+
+9\. **Improve NRIC validation**
 
+  The current NRIC validation used in MediBase3 does not align with Singapore's checksum algorithm for NRICs. As such, MediBase3 does not check if the starting letter and starting 2 digits of the NRIC aligns with the given date of birth.
+  This allows users to enter either incorrect date of births or NRICs that do not align with each other. We plan to enhance the NRIC validation to check for such discrepancies, specifically for patients born on or after 1 January 1968 where [this](https://en.wikipedia.org/wiki/National_Registration_Identity_Card#Structure_of_the_NRIC_number/FIN) rule applies.
```
Comment (PresleyChew) [inline docs/DeveloperGuide.md:1162]: Good!

**Review by PresleyChew — COMMENTED** [review]
LGTM, just a nit

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (TechRaj) [reply docs/DeveloperGuide.md:1162]: we are basically lawyers rather then SWE app HAHAH

**Pull request merged**